python/source/data.rst

```diff
@@ -234,4 +234,57 @@ that match our predicate
     7,
     8,
     9
-  ]
\ No newline at end of file
+  ]
+
+Filtering Arrays for values matching a mask filter
+==================================================
```

Comment:
```suggestion
Filtering Arrays using a mask
=============================
```
The old wording is ok too but "values matching a mask filter" can sound like the values in the array need to match the values in the mask.